src/kernel/vga.zig

```diff
@@ -0,0 +1,483 @@
+/// TODO.
+const REG_START_HORIZONTAL_RETRACE_PULSE: u8  = 0x04;
+/// TODO.
+const REG_END_HORIZONTAL_RETRACE_PULSE: u8    = 0x05;
+/// TODO.
+const REG_VERTICAL_TOTAL: u8                  = 0x06;
+/// TODO.
+const REG_OVERFLOW: u8                        = 0x07;
+/// TODO.
+const REG_PRESET_ROW_SCAN: u8                 = 0x08;
+/// TODO.
+const REG_MAXIMUM_SCAN_LINE: u8               = 0x09;
+/// The command for setting the start of the cursor scan line.
+const REG_CURSOR_START: u8                    = 0x0A;
+/// The command for setting the end of the cursor scan line.
+const REG_CURSOR_END: u8                      = 0x0B;
+/// TODO.
+const REG_START_ADDRESS_HIGH: u8              = 0x0C;
+/// TODO.
+const REG_START_ADDRESS_LOW: u8               = 0x0D;
+/// The command for setting the upper byte of the cursor's linear location.
+const REG_CURSOR_LOCATION_HIGH: u8            = 0x0E;
+/// The command for setting the lower byte of the cursor's linear location.
+const REG_CURSOR_LOCATION_LOW: u8             = 0x0F;
+/// TODO.
+const REG_VERTICAL_RETRACE_START: u8          = 0x10;
+/// TODO.
+const REG_VERTICAL_RETRACE_END: u8            = 0x11;
+/// TODO.
+const REG_VERTICAL_DISPLAY_ENABLE_END: u8     = 0x12;
+/// TODO.
+const REG_OFFSET: u8                          = 0x13;
+/// TODO.
+const REG_UNDERLINE_LOCATION: u8              = 0x14;
+/// TODO.
+const REG_START_VERTICAL_BLINKING: u8         = 0x15;
+/// TODO.
+const REG_END_VERTICAL_BLINKING: u8           = 0x16;
+/// TODO.
+const REG_CRT_MODE_CONTROL: u8                = 0x17;
+/// TODO.
+const REG_LINE_COMPARE: u8                    = 0x18;
+
+
+///The start of the cursor scan line, the very beginning.
+const CURSOR_SCANLINE_START: u8   = 0x0;
+///The scan line for use in the underline cursor shape.
+const CURSOR_SCANLINE_MIDDLE: u8  = 0xE;
+///The end of the cursor scan line, the very end.
+const CURSOR_SCANLINE_END: u8     = 0xF;
+
+/// If set, disables the cursor.
+const CURSOR_DISABLE: u8    = 0x20;
+
+pub const WIDTH: u16        = 80;
+pub const HEIGHT: u16       = 25;
+
+// The set of colours that VGA supports and can display for the foreground and background.
+pub const COLOUR_BLACK: u4            = 0x00;
+pub const COLOUR_BLUE: u4             = 0x01;
+pub const COLOUR_GREEN: u4            = 0x02;
+pub const COLOUR_CYAN: u4             = 0x03;
+pub const COLOUR_RED: u4              = 0x04;
+pub const COLOUR_MAGENTA: u4          = 0x05;
+pub const COLOUR_BROWN: u4            = 0x06;
+pub const COLOUR_LIGHT_GREY: u4       = 0x07;
+pub const COLOUR_DARK_GREY: u4        = 0x08;
+pub const COLOUR_LIGHT_BLUE: u4       = 0x09;
+pub const COLOUR_LIGHT_GREEN: u4      = 0x0A;
+pub const COLOUR_LIGHT_CYAN: u4       = 0x0B;
+pub const COLOUR_LIGHT_RED: u4        = 0x0C;
+pub const COLOUR_LIGHT_MAGENTA: u4    = 0x0D;
+pub const COLOUR_LIGHT_BROWN: u4      = 0x0E;
+pub const COLOUR_WHITE: u4            = 0x0F;
+
+/// The set of shapes that can be displayed.
+pub const CURSOR_SHAPE = enum(u1) {
```

Comment:
I think it's conventional for types to be camelcase, like `CursorShape`.

Reply:
Yes, that is the case